applications/ULFapplication/custom_processes/remove_and_save_wall_process.h

```diff
@@ -0,0 +1,322 @@
+//
+//   Project Name:        Kratos
+//   Last Modified by:    $Author: anonymous $
+//   Date:                $Date: 2007-11-06 12:34:26 $
+//   Revision:            $Revision: 1.4 $
+//
+//  this process save structural elements in a separate list
+
+#if !defined(KRATOS_REMOVE_SAVE_WALL_PROCESS_INCLUDED )
+#define  KRATOS_REMOVE_SAVE_WALL_PROCESS_INCLUDED
+
+
+
+// System includes
+#include <string>
+#include <iostream>
+#include <algorithm>
+
+// External includes
+
+
+// Project includes
+#include "includes/define.h"
+#include "processes/process.h"
+#include "includes/node.h"
+#include "includes/element.h"
+#include "includes/model_part.h"
+//#include "custom_elements/updated_lagrangian_fluid.h"
+//#include "custom_elements/updated_lagrangian_fluid3D.h"
+//#include "custom_elements/updated_lagrangian_fluid_inc.h"
+//#include "custom_elements/updated_lagrangian_fluid3D_inc.h"
+
+
+namespace Kratos
+{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/// Short class definition.
+/** Detail class definition.
+	Update the PRESSURE_FORCE on the nodes
+
+
+*/
+
+class RemoveAndSaveWallNodesProcess
+    : public Process
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of PushStructureProcess
+    KRATOS_CLASS_POINTER_DEFINITION(RemoveAndSaveWallNodesProcess);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructor.
+    RemoveAndSaveWallNodesProcess()
+    //ModelPart& fluid_model_part, ModelPart& structure_model_part, ModelPart& combined_model_part)
+    //: mr_fluid_model_part(fluid_model_part), mr_structure_model_part(structure_model_part), mr_combined_model_part(combined_model_part)
+    {
+	KRATOS_WATCH(" INSIDE REMOVE AND SAVE WALL CONSTRUCTOR") 
+    }
+
+    /// Destructor.
+    virtual ~RemoveAndSaveWallNodesProcess()
+    {
+    }
+
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    //	void operator()()
+    //	{
+    //		MergeParts();
+    //	}
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    void RemoveAndSave(ModelPart& fluid_model_part, ModelPart& wall_model_part)
+    {
+        KRATOS_TRY
+	ModelPart fluid_only_model_part;
+
+        fluid_only_model_part.Elements().clear();
```

Comment:
You should use the Remove methods of modelpart instead of clearing directly the arrays. This will make the submodelparts in inconsistent situation!!